go/src/koding/config/genconfig.go

```diff
@@ -0,0 +1,233 @@
+// +build ignore
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+)
+
+var (
+	input  = flag.String("i", "-", "")
+	output = flag.String("o", "-", "")
+)
+
+var t = template.Must(template.New("").Parse(`// This file is auto-generated. DO NOT EDIT!
```

Comment:
I'm not a big fan of auto-generation but if there isn't any way to achieve correctness and maintainability in golang like you've explained then it sounds good to me.